types/convert.go

```diff
@@ -216,20 +216,30 @@ func convertScientificNotation(str string) (string, error) {
 		} else if point+int(exp) < len(f)-1 { // 123.456 >> 2 = 12345.6
 			return f[:point] + f[point+1:point+1+int(exp)] + "." + f[point+1+int(exp):], nil
 		}
+
+		if point == len(f) { // no point 1E5
+			return f[:point] + strings.Repeat("0", int(exp)), nil
+		}
+
 		// 123.456 >> 5 = 12345600
 		return f[:point] + f[point+1:] + strings.Repeat("0", point+int(exp)-len(f)+1), nil
 	} else { // move point left
 		exp = -exp
 		if int(exp) < point { // 123.456 << 2 = 1.23456
 			return f[:point-int(exp)] + "." + f[point-int(exp):point] + f[point+1:], nil
```

Comment:
Will `123E-1` be handled correctly?

Reply:
I think we could firstly extract the integer part and the decimal part of `f`, then do the point movement according to `exp`, which makes code more understandable.